lib/rules/no-else-return.js

```diff
@@ -221,13 +231,40 @@ module.exports = {
             }
         }
 
+        /**
+         * Check the if statement
+         * @returns {void}
+         * @param {Node} node The node for the if statement to check
+         * @private
+         */
+        function IfStatement(node) {
+            const parent = context.getAncestors().pop();
```

Comment:
Can this line be replaced with `const parent = node.parent`?

Reply:
hmm that breaks stuff

Reply:
I think the cause of the breakage was https://github.com/eslint/eslint/pull/9229#discussion_r141507910, not the `node.parent` change.